### ISSUE TITLE
Traditional curve/duration animation

### ISSUE THREAD

**Issue by twobit** [chenglou/react-motion#68]
Hi Chenglou,

Here are my thoughts after integrating the `Spring` API yesterday:

 * Users of my components will have trouble understanding stiffness/damping compared to the slightly more intuitive and common curve/duration.
 * In some instances the `Spring` animation doesn't return to a stable state.
 * Would be great to see some predefined constants like you get with most curve animation libraries (e.g. stiff_spring, wobbly_spring).

I love the API design but would really like to have the choice to use a Tween object. I know it's got issues like you mentioned in the React-Europe talk but I don't think it would hurt to provide a traditional approach to animation as well.

```
let Demo = React.createClass({
  getInitialState() {
    return {open: false};
  },

  handleMouseDown() {
    this.setState({open: !this.state.open});
  },

  render() {
    return (
      <div>
        <button onMouseDown={this.handleMouseDown}>Toggle</button>
        <Tween endValue={{val: this.state.open ? 400 : 0, config: [EasingTypes.easeInOutQuad, 2]}}>
          {interpolated =>
            <div className="demo0-block" style={{
              transform: `translate3d(${interpolated.val}px, 0, 0)`,
            }} />
          }
        </Tween>
      </div>
    );
  }
});
```

**Comment by chenglou** [comment]
I do plan to document a certain set of configs such as the default one and `[120, 17]` (bouncy).

What do you mean by stable state? Is this related to #31?

The API works for normal tweens, but as soon as you try some interruptible stuff the curve begins to look weird. I was thinking of taking [tween-state](https://github.com/chenglou/react-tween-state)'s additive animation to solve this. But then again you could just use tween-state for now =P.

**Comment by twobit** [comment]
Here's an example of the unstable spring issue:
![image](http://fat.gfycat.com/FlakyResponsibleAfricanparadiseflycatcher.gif)

Definitely don't want to go back to tween-state since we've migrated to ES6 style classes. Regardless, thanks for your work on these libraries.

**Comment by chenglou** [comment]
@iclanzan your PR fixes this right?

**Comment by razh** [comment]
@iclanzan 's PR should resolve almost all issues regarding stability even given the frame-rate changes during resizing. @twobit Can you try out #57?

rebound-js has converters from Origami (where `k` and `b` are represented as `tension` and `friction`) and Pop (`bounciness` and `speed`): http://facebook.github.io/rebound-js/docs/rebound.html#section-61

**Comment by chenglou** [comment]
I feel like having @twobit's suggested `stiff_spring` and `wobbly_spring` and such solves point 1? No trying to dissuade my porting tween-state over, but it seems this can be delayed quite a bit if we do provide convenient named configurations (even more so that they're literally an array of 2 elements that you can inspect and play around with).

Unrelated: react-motion at first almost used rebound-js. Then I realized I only needed a tiny chunk of the library so I just wrote the  `stepper.js` code instead.

**Comment by chenglou** [comment]
I feel like having @twobit's suggested stiff_spring and wobbly_spring and such solves point 1? No trying to dissuade my porting tween-state over, but it seems this can be delayed quite a bit if we do provide convenient named configurations (even more so that they're literally an array of 2 elements that you can inspect and play around with).

Unrelated: react-motion at first almost used rebound-js. Then I realized I only needed a tiny chunk of the library so I just wrote the stepper.js code instead.

Unrelated 2: oh and this library does animate arbitrary data structures now, of course. cc @andreaferretti who mentioned this in chenglou/react-tween-state#22

**Comment by iclanzan** [comment]
Yes, #57 should solve this.

**Comment by twobit** [comment]
#57 fixes the issue for me. Re Unrelated 2, really nice to see that functionality baked into the library :)

**Comment by threepointone** [comment]
@twobit regarding your first concern - would you prefer an api like https://github.com/threepointone/react-ease? I've been meaning to port it over to react-motion, if possible.

**Comment by twobit** [comment]
@threepointone That looks pretty close but I rely on the object interpolation that Spring is doing

**Comment by chenglou** [comment]
@twobit we've addressed point 2 already. I've made a demo to address point 1: https://cdn.rawgit.com/chenglou/react-motion/2a5f89bee700c6bbb0560eee0c8f9ad66d111bdd/demos/demo5/index.html

Have fun with it =)

**Comment by threepointone** [comment]
Yay!

**Comment by chenglou** [comment]
Alright, some carefully chosen constants are now exposed under `constants` [here](https://github.com/chenglou/react-motion/blob/372446fc8679dcdbf4de2e983d53709ab40ce88c/src/constants.js) and will go in the next release (the one after 0.2.2). This, along with that demo I linked above, should ease the usage of spring constants enough for me to deem this issue solved.

As for traditional tweens, @threepointone or I might port the stuff over in the future (I also remember hitting a roadblock with using tween with this API format but that might be wrong). Somewhat low-pri right now because I've got some cooler stuff to show =).
